packages/button/src/Button/Button.tsx

```diff
@@ -13,7 +13,8 @@ export type ButtonProps = GenericProps &
     primary?: boolean
     destructive?: boolean
     secondary?: boolean
-    form?: boolean
```

Comment:
I don't see an extension of HTML Attributes or anything to support adding the real `form` prop

Reply:
Nope, you're absolutely right, that's completely missing - I probably should've made this a draft PR -  just wanted to get this up as a save point yesterday 😉 Will add that in this morning and fix the ts issues 👍

Reply:
Our button props are pretty gnarly, do you think it's worth just extending the whole of the button HTML attributes? Or just picking the form-related one for now 🤔

Reply:
Probably form related one for now

Reply:
At the very least all these look handy to have if they're planning on using the button to submit forms 🤔 
 ![Screenshot 2023-05-05 at 9 36 32 am](https://user-images.githubusercontent.com/36558508/236351947-277536c3-5aa6-40f5-8324-ff835ca05b45.png)

Reply:
> Probably form related one for now
Fair, may be a very large thread to pull at if we were to start refactoring the button type 😆

Reply:
Let's stick with `form` first

Reply:
What are the chances of us just removing the size difference altogether? 😏 It's a breaking change for the consumers anyway, and if it's an anti-pattern maybe we can just say they can't use the smaller one anymore.

Reply:
Because it's a valid use case, without an intuitive way to communicate yet:
https://github.com/cultureamp/kaizen-discourse/issues/50#issuecomment-1447222226

Reply:
Stay tuned for a `SubmitButton` solution preview lol

Reply:
But... what if they just have a bigger button? Would the design explode?

Reply:
Probably not, but it could also be used for tables where the information could be dense, so introducing whitespace across the row could be undesirable. There's been no design discovery done on how small small should be yet, just that this is to mitigate the existing issue whilst giving us the option to improve it further down the road.

Reply:
Ah... well we'd probably need to note that people should refrain from using it in the meantime until we've figured out what it's meant to be 😬

Reply:
Designers actually don't know this exists, so no problem on that part! For now it's purely for backwards compatibility with some snowflake designs.